### ISSUE TITLE
[Investigation] Quantized models support

### ISSUE THREAD

**Issue by huningxin** [intel/webml-polyfill#650]
We need to investigate quantized model support for WebNN API.

Some TODOs in my mind:
1. Support quantized version of operations in WASM backend
2. Modify TFLite model loader to support quantized model.
3. Test [quantized model](https://www.tensorflow.org/lite/guide/hosted_models) (e.g.  Mobilenet_V2_1.0_224_quant) of TFLite model zoo.
4. Investigate the gaps in Chromium POC, starting from Android NNAPI backend.

**Comment by Wenzhao-Xiang** [comment]
I first supported quantized version of Conv2D, DepthwiseConv2D, Averagepool, Reshape and Softmax in WASM backend, and modified the TFLite model loader to support quantized model. Then I tried Mobilenet_V2_1.0_224_quant model for image classification. And here is the result:

**Quantized model:**
![image](https://user-images.githubusercontent.com/40754037/54541593-2c345680-49d5-11e9-8ee4-637e649013c8.png)

**Float model:**
![image](https://user-images.githubusercontent.com/40754037/54541769-7fa6a480-49d5-11e9-8005-886676bb07da.png)


The class seems to be right, but still need to do some post-process with the probability.
Another strange thing is that quantized model is slower than float model. Maybe I need some quantized version of test case to test if the ops work as expect.

**Comment by huningxin** [comment]
Good progress! Please refer to TFLite demo code for post-processing. We need to make sure the result is correct at first step. Thanks!

**Comment by Wenzhao-Xiang** [comment]
Found Mobilenet_V2_1.0_224_quant model have no softmax, so add it automatically.
After post-process, get the following result:

![image](https://user-images.githubusercontent.com/40754037/54577497-0b9ce880-4a37-11e9-8b68-8bed3a5311ee.png)

**Comment by huningxin** [comment]
That's great, thanks @Wenzhao-Xiang !

We need to verify two aspects:
1. what's the result when running on WebNN/NNAPI? @BruceDai , please help on this.
2. what's the reason that quantized model runs slower than float in WASM ops?

**Comment by Wenzhao-Xiang** [comment]
@huningxin 
**Test Env:**
Chromium Version: nightly build 70.0.3503.0 (a7c5589)
Platform: Android 9.0(Google Pixel 2XL)

**Quantized model (Mobilenet_v2):**
![image](https://user-images.githubusercontent.com/40754037/54655670-bed01500-4afd-11e9-981b-b5408a217d47.png)

**Float model (Mobilenet_v2):**
![image](https://user-images.githubusercontent.com/40754037/54655303-6a786580-4afc-11e9-8f0b-e145860ba6f4.png)

Have about **2x speed** up on WebNN/NNAPI.

I will next investigate why quantized model runs slower than float in WASM ops.

**Comment by huningxin** [comment]
The float model number seems slower than what we collected previously. @Wenzhao-Xiang , could you please double-check with @BruceDai ?

And according to [ai benchmark](http://ai-benchmark.com/ranking.html), pixel 3 has significant speedup on int8 quantized model. Please help test on that device. Thanks!

**Comment by Wenzhao-Xiang** [comment]
@huningxin 
Test with benchmark:

Chromium Version: nightly build 70.0.3503.0 (a7c5589)
Platform: Android 9.0(**Google Pixel 2XL**)
**Quantized** Mobilenet_V2:
Inference Time: **79.18+-25.28** [ms]
**Float** Mobilenet_V2:
Inference Time: **109.41+-21.64** [ms]

Chromium Version: nightly build 70.0.3503.0 (a7c5589)
Platform: Android 9.0(**Google Pixel 3**)
**Quantized** Mobilenet_V2:
Inference Time: **8.89+-1.25** [ms]
**Float** Mobilenet_V2:
Inference Time: **105.37+-20.90** [ms]

**Summary**:
About **1.38x speed up** in Google Pixel 2XL
About **10x speed up** in Google Pixel 3

**Comment by Wenzhao-Xiang** [comment]
Also supported ssd_mobilenet_v1_quant for object detection.
Here is the test with examples:
(**Google Pixel 3**)
**Image Classification**:
![image](https://user-images.githubusercontent.com/40754037/54736710-6b39f600-4be7-11e9-9c2b-40dd96dd01fc.png)

**Object Detection**:
![image](https://user-images.githubusercontent.com/40754037/54736694-565d6280-4be7-11e9-9d6a-5030a68502fe.png)

Really have a significant speedup with Google Pixel 3, even faster than WebGL backend with 1080TI, amazing me!

**Comment by huningxin** [comment]
Impressed! Great job @Wenzhao-Xiang ! 

Please follow up as we discussed:
1. publish a demo URL for testing (host both f32 model and int8 model). 
2. investigate the WASM polyfill support
3. make a PR to support int8 model and WASM polyfill

Thanks!

**Comment by Wenzhao-Xiang** [comment]
Please go to [**here**](https://wenzhao-xiang.github.io/intel/webml-examples/examples/image_classification/index.html?prefer=none&b=WASM&m=none&t=none&s=image&d=0) for **image classification demo**(both f32 model and uint8 model, but not include inception v3/v4 and inception resnet v2).

Please go to [**here**](https://wenzhao-xiang.github.io/intel/webml-examples/examples/object_detection/index.html?prefer=none&b=WASM&m=none&t=none&s=image&d=0) for **object detection demo**(both f32 model and uint8 model).

**Now the quantized models are only supported on  WASM and NNAPI backend.**
